app-codepipeline/vars.tf

```diff
@@ -72,24 +72,13 @@ variable "gh_codestar_conn_name" {
   }
 }
 
-variable "cloudinary_name" {
-  type        = string
-  description = "The Cloudinary cloud name"
-  validation {
-    # Not comprehsnsive. Only checks it's non empty.
-    condition     = can(regex("^.+$", var.cloudinary_name))
-    error_message = "Cloudinary name can't be empty."
-  }
-  default = ""
-}
+variable "build_env_vars" {
+  type = list(object({
```

Comment:
This totally works, but see [here](https://github.com/metrotranscom/doorway-infra/blob/377101326958b3601bc0adc43d43f35178799037/bloom-instance/ecs/task/ecs.tf#L32) for an example of how to turn key=>value pairs into a list of `{name: "", value: ""}` objects.